src/pyseekdb/client/embedding_function.py

```
@@ -128,69 +139,378 @@ def dimension_of(embedding_function: EmbeddingFunction[D]) -> int:
+        """
+        if self._preferred_providers is None or len(self._preferred_providers) == 0:
+            if len(self.ort.get_available_providers()) > 0:
+                logger.debug(
+                    f"WARNING: No ONNX providers provided, defaulting to available providers: "
+                    f"{self.ort.get_available_providers()}"
+                )
+            self._preferred_providers = self.ort.get_available_providers()
+        elif not set(self._preferred_providers).issubset(set(self.ort.get_available_providers())):
+            raise ValueError(
+                f"Preferred providers must be subset of available providers: {self.ort.get_available_providers()}"
+            )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)